src/parser.h

```diff
@@ -119,6 +119,10 @@ typedef struct {
 // our encoding and use it to parse identifiers.
 typedef yp_encoding_t *(*yp_encoding_decode_callback_t)(const char *name, size_t width);
 
+typedef struct {
+  bool forwarding;
+} yp_state_t;
```

Comment:
I just happened to see this comment in my inbox and I think in MRI ... will end up adding 3 local variables (*, **, & - idFWD_REST, idFWD_KWREST, and ifFWD_BLOCK).   As special local variables: 

```c
yp_token_list_append(&parser->current_scope->as.scope.locals, &name);
```

Deeper constructs can still see these because local variable lookup is lexical and you have a path to check for it.  I don't know if @kddnewton wants you to do this with 3 variables like MRI or may just make on explicit one but leveraging local variables solves the nesting issue he mentions.

Reply:
Having thought about this overnight I think if there is any goal for syntactic round tripping it should just be one special lvar called '...'.  Using 3 is just overloading the semantics with bare *,**,&.  For JRuby, it doesn't matter either way other than needing to still logically equate the two forms as the same behavior.